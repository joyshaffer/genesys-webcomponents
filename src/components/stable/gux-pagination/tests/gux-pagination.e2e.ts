import { E2EPage, newE2EPage } from '@stencil/core/testing';
import { a11yCheck } from '../../../../../tests/e2eTestUtils';

const axeExclusions = [
  {
    issueId: 'duplicate-id',
    exclusionReason:
      'Test uses seeded value for Math.random, so duplicate ids are expected'
  }
];

async function newNonrandomE2EPage({
  html
}: {
  html: string;
}): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(html);
  await page.waitForChanges();
  await page.addScriptTag({
    path: 'node_modules/axe-core/axe.min.js'
  });
  await page.waitForChanges();

  return page;
}

describe('gux-pagination', () => {
  describe('#render', () => {
    [
      { currentPage: 1, totalItems: 1000, itemsPerPage: 25, layout: 'full' },
      { currentPage: 1, totalItems: 1000, itemsPerPage: 50, layout: 'full' },
      { currentPage: 1, totalItems: 1000, itemsPerPage: 75, layout: 'full' },
      { currentPage: 1, totalItems: 1000, itemsPerPage: 100, layout: 'full' },
      { currentPage: 1, totalItems: 1000, itemsPerPage: 25, layout: 'full' },
      { currentPage: 10, totalItems: 1000, itemsPerPage: 25, layout: 'full' },
      { currentPage: 10, totalItems: 1000, itemsPerPage: 50, layout: 'full' },
      { currentPage: 10, totalItems: 1000, itemsPerPage: 75, layout: 'full' },
      { currentPage: 10, totalItems: 1000, itemsPerPage: 100, layout: 'full' },
      { currentPage: 1, totalItems: 1000, itemsPerPage: 25, layout: 'small' }
    ].forEach(({ currentPage, totalItems, itemsPerPage, layout }, index) => {
      it(`should render as expected (${index + 1})`, async () => {
        const html = `
          <gux-pagination lang="en"
            current-page="${currentPage}"
            total-items="${totalItems}"
            items-per-page="${itemsPerPage}"
            layout="${layout}"
          ></gux-pagination>
        `;
        const page = await newNonrandomE2EPage({ html });
        const element = await page.find('gux-pagination');
        await a11yCheck(page, axeExclusions);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
