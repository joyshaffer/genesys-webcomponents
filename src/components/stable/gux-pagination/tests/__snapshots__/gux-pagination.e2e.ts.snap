// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-pagination #render should render as expected (1) 1`] = `"<gux-pagination lang=\\"en\\" current-page=\\"1\\" total-items=\\"1000\\" items-per-page=\\"25\\" layout=\\"full\\" class=\\"hydrated\\"></gux-pagination>"`;

exports[`gux-pagination #render should render as expected (2) 1`] = `"<gux-pagination lang=\\"en\\" current-page=\\"1\\" total-items=\\"1000\\" items-per-page=\\"50\\" layout=\\"full\\" class=\\"hydrated\\"></gux-pagination>"`;

exports[`gux-pagination #render should render as expected (3) 1`] = `"<gux-pagination lang=\\"en\\" current-page=\\"1\\" total-items=\\"1000\\" items-per-page=\\"75\\" layout=\\"full\\" class=\\"hydrated\\"></gux-pagination>"`;

exports[`gux-pagination #render should render as expected (4) 1`] = `"<gux-pagination lang=\\"en\\" current-page=\\"1\\" total-items=\\"1000\\" items-per-page=\\"100\\" layout=\\"full\\" class=\\"hydrated\\"></gux-pagination>"`;

exports[`gux-pagination #render should render as expected (5) 1`] = `"<gux-pagination lang=\\"en\\" current-page=\\"1\\" total-items=\\"1000\\" items-per-page=\\"25\\" layout=\\"full\\" class=\\"hydrated\\"></gux-pagination>"`;

exports[`gux-pagination #render should render as expected (6) 1`] = `"<gux-pagination lang=\\"en\\" current-page=\\"10\\" total-items=\\"1000\\" items-per-page=\\"25\\" layout=\\"full\\" class=\\"hydrated\\"></gux-pagination>"`;

exports[`gux-pagination #render should render as expected (7) 1`] = `"<gux-pagination lang=\\"en\\" current-page=\\"10\\" total-items=\\"1000\\" items-per-page=\\"50\\" layout=\\"full\\" class=\\"hydrated\\"></gux-pagination>"`;

exports[`gux-pagination #render should render as expected (8) 1`] = `"<gux-pagination lang=\\"en\\" current-page=\\"10\\" total-items=\\"1000\\" items-per-page=\\"75\\" layout=\\"full\\" class=\\"hydrated\\"></gux-pagination>"`;

exports[`gux-pagination #render should render as expected (9) 1`] = `"<gux-pagination lang=\\"en\\" current-page=\\"10\\" total-items=\\"1000\\" items-per-page=\\"100\\" layout=\\"full\\" class=\\"hydrated\\"></gux-pagination>"`;

exports[`gux-pagination #render should render as expected (10) 1`] = `"<gux-pagination lang=\\"en\\" current-page=\\"1\\" total-items=\\"1000\\" items-per-page=\\"25\\" layout=\\"small\\" class=\\"hydrated\\"></gux-pagination>"`;
